src/runtime_src/core/pcie/driver/linux/xocl/lib/libxdma.c

```diff
@@ -941,13 +941,15 @@ static int process_completions(struct xdma_engine *engine,
 
 		if (req->sw_desc_cnt == req->desc_completed) {
 			list_del(&req->entry);
+			spin_unlock(&engine->req_list_lock);
 			if (req->cb && req->cb->io_done) {
 				struct xdma_io_cb *cb = req->cb;
 
-				cb->io_done((unsigned long)cb->private, 0);
+				cb->io_done((unsigned long)cb->private, req->done);
```

Comment:
the 2nd parameter in io_done() is error code, please keep it. If you want additional information on the done bytes, please add a another parameter for it.

Reply:
Hi Karen, I am mostly fine with adding another parameter to io_done function. Before that can we also consider 2 other options:
1. Use the count field in xdma_io_cb to load the done bytes count. It does not seem to be used now. 
2. If the count field is to denote requested len, have a new field "done_bytes" for completed transfers.
3. Rename count to "done_bytes" to ensure that this field will contain completed transfers.

Reply:
Yes, let's do #2. Thanks.